app/src/main/java/uz/direction/coroutineshomework/MemesService.kt

```diff
@@ -1,38 +1,35 @@
 package uz.direction.coroutineshomework
 
 import android.content.Context
+import android.util.Log
 import com.google.gson.Gson
-import com.google.gson.JsonSyntaxException
-import java.io.IOException
+import kotlinx.coroutines.*
 import java.io.InputStream
 import java.nio.charset.Charset
-import kotlin.concurrent.thread
 import kotlin.random.Random
 
 class MemesService {
-
-    fun subscribeToMemes(memesListener: MemesListener, context: Context) {
-        thread {
-            val jsonString: String = try {
-                val inputStream: InputStream = context.assets.open("facts.json")
-                val size: Int = inputStream.available()
-                val buffer = ByteArray(size)
-                inputStream.read(buffer)
-                inputStream.close()
-                String(buffer, Charset.defaultCharset())
-            } catch (e: IOException) {
-                memesListener.onError(e)
-                ""
-            }
-
-            try {
-                val memes = Gson().fromJson(jsonString, Array<Meme>::class.java)
-                val randomIndex = Random.nextInt(memes.size - 1)
-                memesListener.onSuccess(memes[randomIndex])
-            } catch (e: JsonSyntaxException) {
-                memesListener.onError(e)
-            }
+    val myCoroutineScope = CoroutineScope(
+            Dispatchers.IO +
+                    CoroutineName("MemesServiceCoroutine") +
+                    CoroutineExceptionHandler { _, exception ->
+                        Log.e("EX", "Exception \"$exception\" is caught")
+                    })
+
+    suspend fun subscribeToMemes(context: Context): Meme {
+        val meme = myCoroutineScope.async {
```

Comment:
you don't need scope here as it is suspend function and definitely will run under some scope, only thing to worry about is to change coroutine context to IO while running it as it is heavy operation(kind of)